packages/devp2p/src/rlpx/peer.ts

```diff
@@ -499,7 +501,14 @@ export class Peer extends EventEmitter {
     )
 
     try {
-      obj.protocol._handleMessage(msgCode, body.slice(1))
+      let payload = body.slice(1)
+
+      // Use snappy uncompression if peer supports DevP2P >=v5
+      if ((this._hello?.protocolVersion ?? 0) >= 5) {
```

Comment:
What is this fallback to `0` here respectively what are the conditions? Is this when a peer didn't announce its RLPx version? Is this the still a valid connection which needs to be handled?

Reply:
I'm not knowledgeable enough on all the edge cases to know if this is truly a valid scenario or not, and was driven because Typescript is concerned that `_hello.protocolVersion` could be undefined.  Now that you've called it out, I'm assuming we would never try to send a message to a peer where we didn't have the `_hello` fields set?  If the peer didn't announce a Devp2p protocol version, we certainly would not want to run the snappy compression step but maybe a better question is would we want to send a message at all if we don't have the `protocolVersion` identified.  We certainly need to check if `protocolVersion >= 5`, but we may not need to specifically handle for the case of `undefined` at this point in the code.

Reply:
yes sorry this was a way for me to shortern the logic for `protocolVersion` being typed as `| undefined` but it looks kind of confusing.

@acolytec3's original code:
```typescript
if (this._peer._hello?.protocolVersion && this._peer._hello?.protocolVersion >= 5) {
```

Another way could be to check it beforehand like so:

```typescript
    if (!this._peer._hello?.protocolVersion) {
      return // or throw?
    }
    // Use snappy compression if peer supports DevP2P >=v5
    if (this._peer._hello.protocolVersion >= 5) {
```

We could go more deep and try to remove its optionality before it gets there as well.

Reply:
@ryanio Yeah, I guess this is rather a bit confusing and leads to unnecessary questions, will convert back to the old comparison version, this is more clear to read.

(will directly push here)